app/Commands/Process.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Commands;
+
+use App\Services\TransactionProcessor\TransactionProcessor;
+use App\Services\TransactionReader\TransactionReader;
+use LaravelZero\Framework\Commands\Command;
+
+class Process extends Command
+{
+    /**
+     * The signature of the command.
+     *
+     * @var string
+     */
+    protected $signature = 'process
+        {spreadsheet : Absolute or relative path to the spreadsheet to process}
+        {--test : Run the command in test mode}';
+
+    /**
+     * The description of the command.
+     *
+     * @var string
+     */
+    protected $description = 'Process a spreadsheet of transactions';
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle(TransactionReader $transactionReader, TransactionProcessor $transactionProcessor)
+    {
+        $spreadsheet = $this->argument('spreadsheet');
+
+        assert(is_string($spreadsheet));
+
+        if (! is_file($spreadsheet)) {
+            $this->error(sprintf('No spreadsheet could be found at %s', $spreadsheet));
+
+            return self::INVALID;
+        }
+
+        if (! $this->option('test')) {
+            $this->callSilent('migrate:fresh');
+        }
```

Comment:
ℹ️ This is really ugly, but I couldn't find a way to mock the command so came up with this in the meantime so the tests can run